### PR TITLE
Fix cigroup 1 with dataGridTable helper functions

### LINKED ISSUE #5100: [Data Explorer][Discover][Functional Test] Fix discover related failed functional test in ciGroup1
**Describe the bug**
Fix functional tests relating to context(surrounding docs view)

### PR DESCRIPTION
**Pull request by abbyhu2000**
### Description
Fix all context related functional tests

### Issues Resolved
resolves #5100

### CODE REVIEW

File: test/functional/services/data_grid.ts [hunk 1]
```diff
@@ -96,6 +98,42 @@ export function DataGridProvider({ getService }: FtrProviderContext) {
       await testSubjects.click(`dataGridHeaderCell-${columnName}`);
       await find.clickByButtonText('Remove column');
     }
+
+    async getDataGridTableTimestamp(): Promise<string[]> {
```
Comment (ananzh) [inline test/functional/services/data_grid.ts:102]: So the primary difference between this one and getLastColumn() is the CSS class selector (`euiDataGridRowCell--date` vs. `euiDataGridRowCell--lastColumn`)?

Given this structure, it makes sense to refactor and abstract away the common logic into a single method. What do you think?

Something like 
```
async getDataGridTableSpecificColumnValues(columnSelector: string): Promise<string[]> {
    const table = await find.byCssSelector('.euiDataGrid');
    const $ = await table.parseDomContent();

    const columnValues: string[] = [];
    $.findTestSubjects('dataGridRowCell')
      .toArray()
      .forEach((cell) => {
        const cCell = $(cell);
        if (cCell.hasClass(`euiDataGridRowCell--${columnSelector}`)) {
          // The column structure may be very nested to get the actual text
          columnValues.push(cCell.children().children().children().children().text());
        }
      });

    return columnValues;
}
```

`columnSelector` could be `date` or `lastColumn`.

File: test/functional/apps/context/_context_navigation.js [hunk 2]
```diff
@@ -39,33 +39,37 @@ const TEST_FILTER_COLUMN_NAMES = [
...
+        // click view surrounding documents
+        await testSubjects.click('docTableRowAction-0');
+
+        //navigate to the new window
+        await testSubjects.exists('docTable');
+        await browser.switchTab(1);
+
+        //close the new tab and get back to the old tab
+        await browser.closeCurrentWindow();
+        await browser.switchTab(0);
```
Comment (ashwin-pc) [inline test/functional/apps/context/_context_navigation.js:70]: Thanks for adding the comments here :)

File: test/functional/apps/context/_date_nanos.js [hunk 3]
```diff
@@ -75,17 +75,17 @@ export default function ({ getService, getPageObjects }) {
...
-        'Sep 19, 2015 @ 06:50:13.0001000011',
+        'Sep 22, 2019 @ 23:50:13.253123345',
+        'Sep 18, 2019 @ 06:50:13.000000104',
+        'Sep 18, 2019 @ 06:50:13.000000103',
+        'Sep 18, 2019 @ 06:50:13.000000102',
+        'Sep 18, 2019 @ 06:50:13.000000101',
+        'Sep 18, 2019 @ 06:50:13.000000001',
+        'Sep 18, 2019 @ 06:50:13.000000000',
+        'Sep 18, 2019 @ 06:50:12.999999999',
+        'Sep 19, 2015 @ 06:50:13.000100001',
```
Comment (ashwin-pc) [inline test/functional/apps/context/_date_nanos.js:88]: Do we know why these values changed?

File: test/functional/services/data_grid.ts [hunk 4]
```diff
@@ -40,6 +40,8 @@ export function DataGridProvider({ getService }: FtrProviderContext) {
   const testSubjects = getService('testSubjects');
 
   class DataGrid {
+    // This test no longer works in the new data explorer data grid table
+    // since each data grid table cell is now rendered differently
```
Comment (ashwin-pc) [inline test/functional/services/data_grid.ts:44]: Can you elaborate a bit more about this? Why should that prevent us from reading the data?

File: test/functional/services/data_grid.ts (hunk 1 above)
Comment (ashwin-pc) [reply test/functional/services/data_grid.ts:102]: +1 although id keep it simpler like `getDataGridTableColumn`

**Review by ashwin-pc — COMMENTED** [review]
LGTM, once the comments are addressed this should be good to merge :)

**Comment by ashwin-pc** [comment]
The unit test for CI group 3 is failing here. Looks like you need to update a snapshot

File: test/functional/services/data_grid.ts (hunk 4 above)
Comment (abbyhu2000) [reply test/functional/services/data_grid.ts:44]: This function get the text of each table cell using `cCell.find('.euiDataGridRowCell__truncate').text()`. However, in the new data grid table of data explorer, the data-test-subj of `.euiDataGridRowCell__truncate` no longer exists. 

Also now not every data table cell have the same HTML structure. For example, the timestamp column now has a very nested structure where i can only get the text using `cCell.children().children().children().children().text()`. 

While another column like the `_source` column have a list of <dl> and <dd> elements where we need to traverse through all those elements to get the overall text.

Comment (abbyhu2000) [reply test/functional/services/data_grid.ts:44]: But it does seem like except for the `_source` column, all the other table cells can retrieve their text using `cCell.children().children().children().children().text()`

**Comment by abbyhu2000** [comment]
https://github.com/opensearch-project/OpenSearch-Dashboards/actions/runs/6304441448/job/17115796891?pr=5105 cigroup1 functional test passing

File: test/functional/apps/context/_date_nanos.js (hunk 3 above)
Comment (abbyhu2000) [reply test/functional/apps/context/_date_nanos.js:88]: i had to delete the additional number at the end to make the values matched, i am not sure the cause, but it seems like new data grid table only render 9 digits now? @ananzh

**Review by ananzh — APPROVED** [review]

**Pull request merged**